src/DssProxyActionsStETH.sol

```diff
@@ -0,0 +1,481 @@
+        dai.transferFrom(msg.sender, address(this), wad);
+        // Approves adapter to take the DAI amount
+        dai.approve(daiJoin, wad);
+        // Joins DAI into the vat
+        DaiJoinLike(daiJoin).join(urn, wad);
+    }
+}
+
+contract DssProxyActionsStETH is Common {
+
+    constructor(address vat_, address manager_) public Common(vat_, manager_) {}
+
+    // Internal functions
+
+    function _sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x, "sub-overflow");
+    }
+
+    function _toInt256(uint256 x) internal pure returns (int256 y) {
+        y = int256(x);
+        require(y >= 0, "int-overflow");
+    }
+
+    function _toRad(uint256 wad) internal pure returns (uint256 rad) {
+        rad = _mul(wad, 10 ** 27);
+    }
+
+    function _getDrawDart(
+        address jug,
+        address urn,
+        bytes32 ilk,
+        uint256 wad
+    ) internal returns (int256 dart) {
+        // Updates stability fee rate
+        uint256 rate = JugLike(jug).drip(ilk);
+
+        // Gets DAI balance of the urn in the vat
+        uint256 dai = vat.dai(urn);
+
+        // If there was already enough DAI in the vat balance, just exits it without adding more debt
+        uint256 rad = _mul(wad, RAY);
+        if (dai < rad) {
+            uint256 toDraw = rad - dai; // dai < rad
+            // Calculates the needed dart so together with the existing dai in the vat is enough to exit wad amount of DAI tokens
+            dart = _toInt256(toDraw / rate);
+            // This is neeeded due lack of precision. It might need to sum an extra dart wei (for the given DAI wad amount)
+            dart = _mul(uint256(dart), rate) < toDraw ? dart + 1 : dart;
+        }
+    }
+
+    function _getWipeDart(
+        uint256 dai,
+        address urn,
+        bytes32 ilk
+    ) internal view returns (int256 dart) {
+        // Gets actual rate from the vat
+        (, uint256 rate,,,) = vat.ilks(ilk);
+        // Gets actual art value of the urn
+        (, uint256 art) = vat.urns(ilk, urn);
+
+        // Uses the whole dai balance in the vat to reduce the debt
+        dart = _toInt256(dai / rate);
+        // Checks the calculated dart is not higher than urn.art (total debt), otherwise uses its value
+        dart = uint256(dart) <= art ? - dart : - _toInt256(art);
+    }
+
+    function _getWipeAllWad(
+        address usr,
+        address urn,
+        bytes32 ilk
+    ) internal view returns (uint256 wad) {
+        // Gets actual rate from the vat
+        (, uint256 rate,,,) = vat.ilks(ilk);
+        // Gets actual art value of the urn
+        (, uint256 art) = vat.urns(ilk, urn);
+
+        // Gets DAI balance of the urn in the vat
+        uint256 dai = vat.dai(usr);
+
+        // If there was already enough DAI in the vat balance, no need to join more
+        uint256 debt = _mul(art, rate);
+        if (debt > dai) {
+            uint256 rad = _sub(debt, dai);
```

Comment:
```suggestion
            uint256 rad = debt - dai;
```
No need of safe math here.

Reply:
I think if you change this you will be able to remove `function sub` as well

Reply:
Done.